crates/ruff_python_formatter/src/comments/placement.rs

```diff
@@ -664,6 +675,41 @@ fn handle_trailing_end_of_line_condition_comment<'a>(
     CommentPlacement::Default(comment)
 }
 
+/// Handles end of line comments after the `:` of an except clause
+///
+/// ```python
+/// try:
+///    ...
+/// except: # comment
+///     pass
+/// ```
+///
+/// It attaches the comment as dangling comment to the enclosing except handler.
+fn handle_trailing_end_of_line_except_comment<'a>(
+    comment: DecoratedComment<'a>,
+    _locator: &Locator,
+) -> CommentPlacement<'a> {
+    let AnyNodeRef::ExceptHandlerExceptHandler(handler) = comment.enclosing_node() else {
+        return CommentPlacement::Default(comment);
+    };
+
+    // Must be an end of line comment
+    if comment.line_position().is_own_line() {
+        return CommentPlacement::Default(comment);
+    }
+
+    let Some(first_body_statement) = handler.body.first() else {
+        debug_assert!(false); // handlers must have a body
+        return CommentPlacement::Default(comment);
+    };
+
+    if comment.slice().start() > first_body_statement.range().end() {
+        return CommentPlacement::Default(comment);
+    }
+
+    return CommentPlacement::dangling(comment.enclosing_node(), comment);
+}
+
```

Comment:
would probably be a good idea to have some tests for these functions. not sure what we have available to make tests ergonomic. will have a look around, but you might have good ideas already

Reply:
We have some snapshot tests in the `comments/mod.rs` file, that test if the comments get associated with the right node. But they are quiet similar to the formatting snapshot tests. 

https://github.com/astral-sh/ruff/blob/d3d69a031e3070a892e41b41bc191b6d0d11ea88/crates/ruff_python_formatter/src/comments/mod.rs#L587-L608

I plan to review your PR on Monday.

Reply:
yes i did find those and tried to (ab?)use them to generate input i could pass to the functions here like `handle_trailing_end_of_line_except_comment` but i failed.

Reply:
Yeah. It is testing the whole `place_comment` and not only `handle_treiling_end_of_line_except_comment`. This is intentional because the construction of the `DecoratedComment` is very specific and easy to get wrong when constructing by hand (and we want to make sure that changes the `DecoratedComment` construction resulting in a failure of the `except` comment placement are catched too). But you can tailor your source code so that it calls your function with the arguments that you want to test.